File: framework/languages/java/e2e-kotlin/SpotService/src/main/java/systems/zlink/e2e/kotlin/spotservice/NoopIngressHandler.java
package systems.zlink.e2e.kotlin.spotservice;

import systems.zlink.framework.channels.ZLinkRequestContext;
import systems.zlink.framework.channels.ZLinkRequestHandler;

public final class NoopIngressHandler implements ZLinkRequestHandler<String, String> {
    private final ScenarioState state;

    public NoopIngressHandler(ScenarioState state) {
        this.state = state;
    }

    @Override
    public String handle(
        String request,
        ZLinkRequestContext context) {
        state.record("IngressRequest", "channel", request);
        return request;
    }
}

File: framework/languages/java/e2e-kotlin/SpotService/Server/Play/src/main/kotlin/systems/zlink/e2e/kotlin/spotservice/play/handlers/NoopIngressHandler.kt
package systems.zlink.e2e.kotlin.spotservice.play.handlers

import systems.zlink.e2e.kotlin.spotservice.Contracts
import systems.zlink.e2e.kotlin.spotservice.ScenarioState
import systems.zlink.e2e.kotlin.spotservice.play.spots.*
import systems.zlink.framework.channels.ZLinkRequestContext
import systems.zlink.framework.kotlin.ZLinkSuspendingRequestHandler

class NoopIngressHandler(
    private val state: ScenarioState,
) : ZLinkSuspendingRequestHandler<String, String> {
    override suspend fun handle(request: String, context: ZLinkRequestContext): String {
        state.record("IngressRequest", "channel", request)
        return request
    }
}
